src/internal/v1api/providers.go

```diff
@@ -69,6 +74,16 @@ func (p ProviderGenerator) VersionListing() ProviderVersionListingResponse {
 func (p ProviderGenerator) VersionDetails() map[string]ProviderVersionDetails {
 	versionDetails := make(map[string]ProviderVersionDetails)
 
+	keyCollection := gpg.KeyCollection{
+		Namespace: p.Provider.EffectiveNamespace(),
+		Directory: p.KeyLocation,
+	}
+
+	keys, err := keyCollection.ListKeys()
+	if err != nil {
+		p.log.Error("Failed to list keys", slog.Any("err", err))
```

Comment:
Do we want to return the error?

Reply:
we probably should. I'll add this tomorrow morning. Good catch!